slot.go

```diff
@@ -75,6 +77,43 @@ func (s *Slot) Key(id string) (*Key, error) {
 	})
 }
 
+// GetIDByLabel returns a hex-encoded ID for the given label
+func (s *Slot) GetIDByLabel(label string) (string, error) {
```

Comment:
Do you actually need the `ID`? Or is this just a stepping stone to calling `Key`? If it's the latter, a `KeyFromLabel` function might be an easier to understand API